### PR TITLE
[DO NOT MERGE] Building locally using singularity instead of docker

### PR DESCRIPTION
**Pull request by ngam**
cc @conda-forge/tensorflow if you'd like to provide feedback. Also, thoughts on ways to check if this is working correctly or basically equivalent to the docker procedure?

This generally works fine. But some issues (see below) need fixing and tuning at some point.

---

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

---

Issues:
- [ ] lol: `Auto-Configuration Warning: 'TMP' environment variable is not set, using 'C:\Windows\Temp' as default`
- [ ] how to even verify this? (Should test with a smaller package first)
- [ ] nonfatal; build proceeds fine: `error    libmamba Error opening for writing "/opt/conda/pkgs/urls.txt": Read-only file system`
- [ ] how to better isolate system from mess
- [ ] possible to avoid creating a new env?
- [ ] how to limit CPU_COUNT? `CPU_COUNT=10` doesn't work...

### CODE REVIEW

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -28,6 +31,9 @@ conda-build:
 
 CONDARC
 
+mamba create -n building --yes --quiet
+conda clean --packages --yes --quiet
+source activate building
```
Comment (ngam) [inline .scripts/build_steps.sh:36]: Really need this out...

**Comment by jakirkham** [comment]
These scripts are autogenerated so these changes won't persist. If you are looking to make changes to the scripts, would suggest submitting a PR to [conda-smithy]( https://github.com/conda-forge/conda-smithy ). Though we may need to discuss the changes more (swapping out `docker` for `singularity` generally wouldn't be viable, but perhaps a config value or environment variable for the command might work)

**Comment by ngam** [comment]
> These scripts are autogenerated so these changes won't persist. If you are looking to make changes to the scripts, would suggest submitting a PR to [conda-smithy](https://github.com/conda-forge/conda-smithy). Though we may need to discuss the changes more (swapping out `docker` for `singularity` generally wouldn't be viable, but perhaps a config value or environment variable for the command might work)

Yes, this is just for people who wish to build it locally on machines with no docker --- not a full replacement. Is it okay to explore it as a proof-of-concept first before going to conda-smithy? Has this been explored before to your knoweldge?

**Comment by jakirkham** [comment]
No objection to exploration of course. Just trying to provide some guidance

Not that I'm aware of, but can see why this may be of interest

**Comment by ngam** [comment]
> No objection to exploration of course. Just trying to provide some guidance
> 
> Not that I'm aware of, but can see why this may be of interest

Thank you for spotting this, I had no idea where I would file an issue regarding this anyway :)

The issue is that it is not straightforward to stuff in singularity (compared to docker)... so it is a long shot for me, unless more experienced people figure it out..

**Comment by ngam** [comment]
[7a744d3](https://github.com/conda-forge/tensorflow-feedstock/pull/229/commits/7a744d3bbb13ec01ca4a881ec2cf13ccea283020) works perfectly!


<details>

```
####################################################################################
Resource usage summary:

Total time: 0:55:26.9
CPU usage: sys=0:00:39.0, user=0:01:13.3
Maximum memory usage observed: 1.2G
Total disk usage observed (not including envs): 11.5M
Validating outputs
+ validate_recipe_outputs tensorflow-feedstock
validation results:
{
  "linux-64/libtensorflow-2.7.1-cuda112h98e1010_0.tar.bz2": true,
  "linux-64/libtensorflow_cc-2.7.1-cuda112h98e1010_0.tar.bz2": true,
  "linux-64/tensorflow-base-2.7.1-cuda112py39h6917f46_0.tar.bz2": true,
  "linux-64/tensorflow-estimator-2.7.1-cuda112py39hd320b7a_0.tar.bz2": true,
  "linux-64/tensorflow-2.7.1-cuda112py39h01bd6f0_0.tar.bz2": true,
  "linux-64/tensorflow-gpu-2.7.1-cuda112py39h0bbbad9_0.tar.bz2": true
}

```

</details>

**Pull request closed without merging**